rust/operator-binary/src/controller.rs

```diff
@@ -845,16 +849,123 @@ fn build_node_rolegroup_statefulset(
     })
 }
 
+/// Build the [`Job`](`stackable_operator::k8s_openapi::api::batch::v1::Job`) that creates a
+/// NiFi `ReportingTask` in order to enable JVM and NiFi metrics.
+///
+/// The Job is run via the [`tools`](https://github.com/stackabletech/docker-images/tree/main/tools)
+/// docker image and more specifically the `create_nifi_reporting_task.py` Python script.
+///
+/// This script uses the [`nipyapi`](https://nipyapi.readthedocs.io/en/latest/readme.html)
+/// library to authenticate and run the required REST calls to the NiFi REST API.     
+///
+/// In order to authenticate we need the `username` and `password` from the
+/// [`NifiAuthenticationConfig`](`stackable_nifi_crd::authentication::NifiAuthenticationConfig`)
+/// as well as a public certificate provided by the Stackable
+/// [`secret-operator`](https://github.com/stackabletech/secret-operator)
+///
+fn build_reporting_task_job(
+    nifi: &NifiCluster,
+    rolegroup_ref: &RoleGroupRef<NifiCluster>,
+) -> Result<Job> {
+    let rolegroup_obj_name = rolegroup_ref.object_name();
+    let namespace: &str = &nifi.namespace().context(ObjectHasNoNamespaceSnafu)?;
+    let nifi_connect_url = format!(
+        "https://{rolegroup}-0.{rolegroup}.{namespace}.svc.cluster.local:{port}/nifi-api",
```

Comment:
Maybe getting the URL to the API could be a function on the cluster object? It looks like it could be useful in other places

EDIT: Further down I saw the nifi.node_role_service_fqdn(), maybe that could be used as well?

Reply:
node_role_service_fqdn() does refer to the service. Nifi is a bit tricky of what hosts it accepts, so thats why there is the pod fqdn specified.  I would keep it as is for now, we can put that in a method once it reappears?

Reply:
fine by me